### PR TITLE
Fix GH-127 add support for HATEOAS 

### PR DESCRIPTION
**Pull request by blackleg**
Fix [GH-127](https://github.com/spring-cloud/spring-cloud-openfeign/issues/127), Add support for HATEOAS

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@blackleg Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-openfeign&pullRequestId=140)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-openfeign&pullRequestId=140) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by spencergibb** [comment]
Looking good! Tests will be important.

**Comment by pivotal-issuemaster** [comment]
@blackleg Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-openfeign&pullRequestId=140)!

**Comment by blackleg** [comment]
@ryanjbaxter I added some documentation about how enable Hateoas support. Could you review that?

File: docs/src/main/asciidoc/spring-cloud-openfeign.adoc [hunk 1]
```diff
@@ -487,3 +487,21 @@ public class DemoTemplate {
     String demoEndpoint(@SpringQueryMap Params params);
 }
 ----
+
+=== Hateoas support
+
+If your project use the `org.springframework.boot:spring-boot-starter-hateoas` starter
+or the `org.springframework.boot:spring-boot-starter-data-rest` starter, Hateoas support is enabled by default.
```
Comment (ryanjbaxter) [inline docs/src/main/asciidoc/spring-cloud-openfeign.adoc]: Can you elaborate a little more on what Hateoas support means?

File: docs/src/main/asciidoc/spring-cloud-openfeign.adoc [hunk 2]
```diff
@@ -487,3 +487,21 @@ public class DemoTemplate {
...
+
+[source,java,indent=0]
+----
+@FeignClient("demo")
+public class DemoTemplate {
```
Comment (ryanjbaxter) [inline docs/src/main/asciidoc/spring-cloud-openfeign.adoc]: This should be `interface` not `class`

**Comment by blackleg** [comment]
@ryanjbaxter I rewrited the documentation. It isn't more elaborate because I don't known it in depth.

**Comment by spencergibb** [comment]
It doesn't look like the actual functionality is tested via feign, just that things are registered.

**Comment by blackleg** [comment]
@spencergibb @ryanjbaxter I added some tests to test the functionality.

**Pull request merged**